Assets/MRTK/Extensions/package.json

```diff
@@ -26,5 +26,6 @@
       },
     "dependencies": {
         "com.microsoft.mixedreality.toolkit.foundation": "%version%"
-    }
+    },
+    %samples%
```

Comment:
If this is going to be a templated package.json, we shouldn't call it package.json (lest we have folks who try to consume it directly using git submoduling who then ask why it's not working).

This should either be:

1. We can rename this to package.json.template and then rename it to package.json as part of the CI packing
2. Get rid of %samples% and have the CI packing just add the samples in (i'm sure we could parse the JSON there, add it it, then reserialize it on output)

Going with #2 would allow for git submoduling, though 1 is probably easier

Reply:
This might feel pedantic... which I guess it partially is. However, package.json as a concept has meaning outside of our changes, and this file doesn't match the meaning of the existing package.json. It's actually a template. So we should call it package.json.template

Reply:
by that definition, all of our package.json files are templates :)

Reply:
In that case, they should be called package.json.template

Otherwise I guarantee you  some N > 2 people out there will try to submodule this and use it as a UPM package via a git address and then probably waste 30 minutes to an hour figuring out why things aren't working.

Reply:
I'd also be interested in replacing the `%version%` references with a valid version number (maybe `2.5.0-preview.0` or something) to allow people to directly reference the package.json files without any additional work. We can build a regex (I'm using one in a different project that might be applicable here) to find the `version: x.x.x` line and replace it. It gets a bit more interesting in the dependencies lines, but I'm sure we can figure it out!

Reply:
i would be ok calling them templates. i could also see replacing all "2.5.0-preview.0" entries with the actually generated value.

Let me finish the JSON investigation and i will see what i can do.

Reply:
renamed "package.json*" to "packagetemplate.json*"